backend-shared/exportfuncs.cpp

```diff
@@ -4,21 +4,17 @@
 #include <QtConcurrent>
 #include "core/membuffer.h"
 #include "core/divesite.h"
+#include "core/gettextfromc.h"
 #include "core/tag.h"
 #include "core/file.h"
 #include "core/errorhelper.h"
 #include "core/divefilter.h"
 #include "core/divesite.h"
 #include "exportfuncs.h"
 
+namespace Export {
```

Comment:
a global singleton pattern seems to be preferred in Subsurface.
`class->instance()->doFunc()`
no strong opinions on my side.

Reply:
Heh. I'm always arguing the other way round. :-P

Note that these functions have no global state -> why would we want an instance and even more strangely an instance that can send/receive signals, i.e. a QObject. The analogy I see is more qthelper.cpp than mainwindow.cpp.

Checking the history - indeed these functions were lifted the `DiveLogExportDialog` class. But note that this class does *not* have an instance method. This makes sense to me: why wouldn't you allow the user to open more than one of these dialogs (on OSs that tend to be non-modal)?

Reply:
maybe these can just be public static functions in a class.

one problem i had with namespaces is that they can be defined in multiple files and if the project is poorly organized one has to `grep` to see what goes in the namespace.

Reply:
My gut feeling would be to simply remove the namespace. A user of these functions `#include`s the header file and just calls them. They are marked with export_ anyway so there should be no risk of name-clash and the names can be `grep`ed.

Reply:
possibly better, yes.

Reply:
I find namespaces frustrating (we have a couple) as they break my pattern of grepping for stuff. In a class at least the .cpp file contains `ClassName::method()` so I can find that easily. In a namespace I can't.
So my preference would be that we keep this a class - but maybe not a `QObject`?
Otherwise I just don't think this adds a lot of value

Reply:
@dirkhh, @neolit123:I am aware that this is idle bike-shedding!

> In a class at least the .cpp file contains ClassName::method() so I can find that easily. In a namespace I can't.

Fair enough. Though I must say that personally I never `grep` for the namespace - `grep`ing for the function itself has always been sufficiently unambiguous.

My take on namespaces: Their use is primarily for libraries of *different* vendors to avoid name-clashes between exported functions. In C you would prefix every function with gtk_, in C++ you define a namespace Gtk. So I don't really see the use for a non-library project. For the undo-commands, I created a namespace for convenience/laziness - I didn't want to prefix functions that outside code isn't supposed to touch anyway with with command_

Here I went the namespace route to make the changes less drastic. However, the functions are prefixed with export anyway, so why bother with a class *or* namespace? As I said further up: If you want to use the export functions, include the appropriate header and just use them.

Has anyone ever complained about all our core-C functions not being stuck in classes? I certainly haven't missed that.

Therefore, I propose to simply remove the class. I have no problem with defining a class full of static functions. But frankly - I don't see the appeal.

Reply:
I'm perfectly happy with not having class or namespace